### ISSUE TITLE
[Bug]: few list examples break axe core rule "nested-interactive"

### ISSUE THREAD

**Issue by kolaps33** [microsoft/fluentui#33257]
### Component

List (Preview)

### Package version

v9.56.1

### React version

18.3.1

### Environment

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (22) x64 Intel(R) Core(TM) Ultra 7 165H
    Memory: 41.66 GB / 63.64 GB
  Browsers:
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.22621.3527
```

### Current Behavior

![Image](https://github.com/user-attachments/assets/7e734d88-8e66-4504-a8f0-d363f52eb380)

### Expected Behavior

There are no axe core scan bugs. 

### Reproduction

https://stackblitz.com/run?file=src%2Fexample.tsx

### Steps to reproduce

1. Go to list preview page: https://react.fluentui.dev/?path=/docs/preview-components-list--docs
2. run "[accessibility insight for web](https://accessibilityinsights.io/downloads/)" extension or [Axe core extension](https://chromewebstore.google.com/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd) in order to scan the page 
3. observe results

Actual: axe core scan throw [nested-interactive](https://dequeuniversity.com/rules/axe/4.9/nested-interactive?application=msftAI) error
Expected: there should be no axe core scan errors 

Note:
These 3 examples are affected:
https://react.fluentui.dev/?path=/docs/preview-components-list--docs#single-action-selection
https://react.fluentui.dev/?path=/docs/preview-components-list--docs#single-action-selection-controlled
https://react.fluentui.dev/?path=/docs/preview-components-list--docs#single-action-selection-different-primary

For each example the issue is causing the checkbox which is basically input.
The reasonable solution would be to change it the same way as menuitem checkboxes has it:
![Image](https://github.com/user-attachments/assets/1f2cd4c5-7c00-4531-8634-50249b3d0a73)

### Are you reporting an Accessibility issue?

None

### Suggested severity

Urgent - No workaround and Products/sites are affected

### Are you willing to submit a PR to fix?

yes

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

**Comment by khmakoto** [comment]
@george-cz fyi, since you worked on `List`

**Comment by george-cz** [comment]
While this is technically a reported bug, the checkbox is just a visual clue, as the List is using a `listbox` with `option` roles, so the user is not expected to go in the list item and focus on the checkbox.
